mne_bids/read.py

```diff
@@ -63,6 +63,77 @@ def _read_raw(raw_fpath, electrode=None, hsp=None, hpi=None,
     return raw
 
 
+def _read_events(events_data, event_id, raw, verbose=None):
+    """Retrieve events (for use in *_events.tsv) from FIFF/array & Annoations.
+
+    Parameters
+    ----------
+    events_data : str | array | None
+        The events file. If a string, a path to the events file. If an array,
+        the MNE events array (shape n_events, 3). If None, events will be
+        inferred from the stim channel using `find_events`.
+    event_id : dict
+        The event id dict used to create a 'trial_type' column in events.tsv,
+        mapping a description key to an integer valued event code.
+    raw : instance of Raw
+        The data as MNE-Python Raw object.
+    verbose : bool | str | int | None
+        If not None, override default verbose level (see :func:`mne.verbose`).
+
+    Returns
+    -------
+    events : array, shape = (n_events, 3)
+        The first column contains the event time in samples and the third
+        column contains the event id. The second column is ignored for now but
+        typically contains the value of the trigger channel either immediately
+        before the event or immediately after.
+    descriptions : dict
+        A dictionary with the keys corresponding to the event descriptions and
+        the values to the event IDs.
+
+    """
+    # get events from events_data
+    if isinstance(events_data, str):
+        events = read_events(events_data, verbose=verbose).astype(int)
+    elif isinstance(events_data, np.ndarray):
+        if events_data.ndim != 2:
+            raise ValueError('Events must have two dimensions, '
+                             f'found {events_data.ndim}')
+        if events_data.shape[1] != 3:
+            raise ValueError('Events must have second dimension of length 3, '
+                             f'found {events_data.shape[1]}')
+        events = events_data
+    else:
+        events = np.empty(shape=(0, 3), dtype=int)
+
+    # Only keep events for which we have an ID <> description mapping.
+    # XXX Should we emit a warning if we drop events?
```

Comment:
throw an error? dropping events without user's knowledge sounds dangerous to me ...

Reply:
Totally agree, see https://github.com/mne-tools/mne-bids/pull/603#issuecomment-716063371

Reply:
I've changed the behavior accordingly.